test/test_pruning_op.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import hypothesis.strategies as st
+from hypothesis import given
+import numpy as np
+import torch
+from torch.testing._internal.common_utils import TestCase
+import torch.testing._internal.hypothesis_utils as hu
+hu.assert_deadline_disabled()
+
+
+class PruningOpTest(TestCase):
+
+    # Generate rowwise mask vector based on indicator and threshold value.
+    # indicator is a vector that contains one value per weight row and it
+    # represents the importance of a row.
+    # We mask a row if its indicator value is less than the threshold.
+    def _generate_rowwise_mask(self, embedding_rows):
+        indicator = torch.from_numpy((np.random.random_sample(embedding_rows)).astype(np.float32))
+        threshold = np.random.random_sample()
+        mask = torch.BoolTensor([True if val >= threshold else False for val in indicator])
+        return mask
+
+    def _test_rowwise_prune_op(self, embedding_rows, embedding_dims, indices_type):
+        weight_types = [np.int8, np.int32, np.int16, np.float32, np.double]
```

Comment:
Can we add this as a hypothesis argument?

Reply:
Also it would be better to add torch types here rather than numpy types.

Reply:
Good points.